de-lib/src/main/java/org/iplantc/de/notifications/client/utils/NotificationUtil.java

```diff
@@ -0,0 +1,272 @@
+     * @return
+     */
+    public NotificationMessage getMessage(Notification notification,
+                                          final NotificationAutoBeanFactory notFactory) {
+
+        NotificationMessage msg = notification.getMessage();
+        msg.setSeen(notification.isSeen());
+        msg.setCategory(NotificationCategory.fromTypeString(notification.getCategory()));
+        Splittable payload = notification.getNotificationPayload();
+
+        if (payload == null) {
+            return msg;
+        }
+
+        switch (msg.getCategory()) {
+            case ALL:
+                GWT.log("ALL Analysis category");
+                break;
+            case APPS:
+                PayloadAppsList appPList =
+                        AutoBeanCodex.decode(notFactory, PayloadAppsList.class, payload).as();
+                if ("share".equalsIgnoreCase(appPList.getAction())) {
+                    msg.setContext(payload.getPayload());
+                } else {
+                    GWT.log("Unhandled apps action type!!");
+                }
+                break;
+            case ANALYSIS:
+                PayloadAnalysis analysisPayload =
+                        AutoBeanCodex.decode(notFactory, PayloadAnalysis.class, payload).as();
+                String analysisAction = analysisPayload.getAction();
+
+                if ("job_status_change".equals(analysisAction) || "share".equals(analysisAction)) {
+                    msg.setContext(payload.getPayload());
+                } else {
+                    GWT.log("Unhandled Analysis action type!!");
+                }
+                break;
+
+            case DATA:
+                PayloadData dataPayload =
+                        AutoBeanCodex.decode(notFactory, PayloadData.class, payload).as();
+                String dataAction = dataPayload.getAction();
+
+                if ("file_uploaded".equals(dataAction)) {
+                    AutoBean<File> fileAb = AutoBeanUtils.getAutoBean(dataPayload.getData());
+                    msg.setContext(AutoBeanCodex.encode(fileAb).getPayload());
+                } else if ("share".equals(dataAction) || "unshare".equals(dataAction)) {
+                    List<String> paths = dataPayload.getPaths();
+                    if (paths != null && !paths.isEmpty()) {
+                        String path = paths.get(0);
+                        Splittable file =
+                                CommonModelUtils.getInstance().createHasPathSplittableFromString(path);
+                        msg.setContext(file.getPayload());
+                    }
+                }
+                break;
+
+            case NEW:
+                GWT.log("NEW  category");
+                break;
+
+            case SYSTEM:
+                GWT.log("SYSTEM  category");
+                break;
+
+            case PERMANENTIDREQUEST:
+                GWT.log("Permanent Id request category");
+                msg.setContext(payload.getPayload());
```

Comment:
I guess we should keep on eye on notifications to make sure they're still working as expected, since this looks slightly different than the code it's replacing in the `NotificationCallbackWrapper`: for example, `PERMANENTIDREQUEST` payloads weren't getting set in that callback, but that might not make a difference or break anything.

Reply:
Good catch.  I did notice (I think it was this PR) that some of the notification filters weren't working so it's possible I fixed one thing but broke something else so I'll double check 👍

Reply:
This was actually a really good catch - because this logic was sort of duplicated in 2 different areas (once for the Notification window, and again for the Notification dropdown on the desktop) there were a couple things that worked in one spot and not the other.  I'll put up a PR for fixes soon.